Terminal.Gui/Views/Menu/MenuBar.cs

```diff
@@ -0,0 +1,1392 @@
+	/// If the user presses any <see cref="MenuItem.HotKey"/>s defined in the <see cref="MenuBarItem"/>s, the menu bar will be activated and the sub-menu will be opened.
+	/// </para>
+	/// <para>
+	/// If the user presses any <see cref="MenuItem.HotKey"/>s defined in the <see cref="MenuBarItem"/>s, the menu bar will be activated and the sub-menu will be opened.
+	/// </para>
+	/// <para>
+	/// <see cref="Key.Esc"/> will close the menu bar and any open sub-menus.
+	/// </para>
+	/// </remarks>
+	public Key Key {
+		get => _key;
+		set {
+			if (_key == value) {
+				return;
+			}
+			ClearKeyBinding (_key);
+			AddKeyBinding (value, Command.Select);
+			_key = value;
+		}
+	}
+
+
+	bool _useKeysUpDownAsKeysLeftRight = false;
+	/// <summary>
+	/// Used for change the navigation key style.
+	/// </summary>
+	public bool UseKeysUpDownAsKeysLeftRight {
+		get => _useKeysUpDownAsKeysLeftRight;
+		set {
+			_useKeysUpDownAsKeysLeftRight = value;
+			if (value && UseSubMenusSingleFrame) {
+				UseSubMenusSingleFrame = false;
+				SetNeedsDisplay ();
+			}
+		}
+	}
+
+	static Rune _shortcutDelimiter = new Rune ('+');
+	/// <summary>
+	/// Sets or gets the shortcut delimiter separator. The default is "+".
+	/// </summary>
+	public static Rune ShortcutDelimiter {
+		get => _shortcutDelimiter;
+		set {
+			if (_shortcutDelimiter != value) {
+				_shortcutDelimiter = value == default ? new Rune ('+') : value;
+			}
+		}
+	}
+
+	/// <summary>
+	/// The specifier character for the hot keys.
+	/// </summary>
+	public new static Rune HotKeySpecifier => (Rune)'_';
+
+	// Set in OnInvokingKeyBindings. -1 means no menu item is selected for activation.
+	int _menuBarItemToActivate;
+	// Set in OnInvokingKeyBindings. null means no sub-menu is selected for activation.
+	MenuItem _menuItemToSelect;
+	bool _openedByAltKey;
+
+	/// <summary>
+	/// Called when a key bound to Command.Select is pressed. Either activates the menu item or runs it, depending on whether it has a sub-menu.
+	/// If the menu is open, it will close the menu bar.
+	/// </summary>
+	/// <returns></returns>
+	bool SelectOrRun ()
+	{
+		if (!IsInitialized || !Visible) {
+			return true;
+		}
+
+		if (_menuBarItemToActivate != -1) {
+			Activate (_menuBarItemToActivate);
+		} else if (_menuItemToSelect != null) {
+			Run (_menuItemToSelect.Action);
+		} else {
+			if (IsMenuOpen && _openMenu != null) {
+				CloseAllMenus ();
+			} else {
+				OpenMenu ();
+			}
+
+		}
+		return true;
+	}
+
+	/// <inheritdoc/>
+	public override bool OnInvokeKeyBindings (KeyEventArgs keyEvent)
+	{
+		// This is a bit of a hack. We want to handle the key bindings for menu bar but
+		// InvokeKeyBindings doesn't pass any context so we can't tell which item it is for.
+		// So before we call the base class we set SelectedItem appropriately.
+		// TODO: Figure out if there's a way to have KeyBindings pass context instead. Maybe a KeyBindingContext property?
+
+		var key = keyEvent.Key;
+
+		if (ContainsKeyBinding (key)) {
```

Comment:
Must prevents for `Key.Null`.

`		if (key != Key.Null && ContainsKeyBinding (key)) {
`